freqtrade/optimize/hyperopt_loss/hyperopt_loss_profit_sortino_spring_efficiency.py

```diff
@@ -0,0 +1,136 @@
+    """
+    Custom loss function for hyperopt optimization combining quadratic profits, Sortino ratio,
+    and hybrid drawdown penalty. The loss function is designed to optimize the trading strategies
+    by prioritizing higher profits and lower drawdowns, while adjusting for risk through Sortino
+    ratio.
+    """
+
+    @staticmethod
+    def hyperopt_loss_function(results: DataFrame, trade_count: int,
+                               min_date: datetime, max_date: datetime,
+                               config: Config, *args, **kwargs) -> float:
+        """
+        Calculate the hyperopt loss based on trade results.
+
+        Parameters:
+        - results: DataFrame containing the trading results
+        - trade_count: int representing the number of trades
+        - min_date: datetime object representing the minimum date in the dataset
+        - max_date: datetime object representing the maximum date in the dataset
+        - config: Config object containing the configuration
+
+        Returns:
+        - The calculated loss as a float
+        """
+
+        try:
+            # Total profit
+            profit = results['profit_abs'].sum()
+
+            # Sortino ratio
+            sortino_ratio = calculate_sortino(
+                results, min_date, max_date, config['dry_run_wallet'])
+
+            # Underwater DataFrame (max value to date, drawdown, drawdown relative)
+            drawdown_df = calculate_underwater(
+                results,
+                value_col='profit_abs',
+                starting_balance=config['dry_run_wallet']
+            )
+
+            # Calculate Max Drawdown
+            max_drawdown = abs(min(drawdown_df['drawdown']))
+
+            # Calculate Average Drawdown (absolute and relative)
+            avg_drawdown, _ = calculate_average_drawdown(drawdown_df)
+
+            # PSSE Loss Function
+            if profit > 0:
+                # Sortino ratio requires at least 2 trades for meaningful results
+                if trade_count < 2:
+                    sortino_ratio = 1
+
+                # Geometric mean of drawdown penalties
+                geometric_drawdown = math.sqrt((1 + max_drawdown) * (1 + avg_drawdown))
+
+                # Harmonic mean of drawdown penalties
+                harmonic_drawdown = (2 / ((1 / (1 + max_drawdown)) + (1 / (1 + avg_drawdown))))
+
+                # Hybrid drawdown penalty (geometric mean combined with harmonic mean)
+                hybrid_drawdown = math.sqrt(
+                    geometric_drawdown * harmonic_drawdown)
+
+                # Combine quadratic profit with Sortino ratio and hybrid drawdown penalty
+                loss = -1 * ((profit ** 2) + (profit * sortino_ratio) / hybrid_drawdown)
+            else:  # profit < 0
+                loss = -profit
+
+            return loss
+
+        # Handle any exception that may occur during loss calculation
+        except (Exception, ValueError):
+            return -profit
```

Comment:
I don't see exception handling improved, really.

It's still running a "catch all" exception - where i would assume that exceptions are actually known (also WHERE they actually happen).